code/modules/hydroponics/grown/nettle.dm

```diff
@@ -105,15 +105,17 @@
 			user.Paralyze(100)
 			to_chat(user, "<span class='userdanger'>You are stunned by [src] as you try picking it up!</span>")
 
-/obj/item/reagent_containers/food/snacks/grown/nettle/death/attack(mob/living/carbon/M, mob/user)
-	if(!..())
-		return
-	if(isliving(M))
-		to_chat(M, "<span class='danger'>You are stunned by the powerful acid of [src]!</span>")
-		log_combat(user, M, "attacked", src)
+/obj/item/reagent_containers/food/snacks/grown/nettle/death/attack(mob/living/M, mob/user)
+	. = ..()
+	if(HAS_TRAIT(user, TRAIT_PACIFISM))
+		return TRUE
+	if(iscarbon(M))
+		var/mob/living/carbon/target = M
+		to_chat(target, "<span class='danger'>You are stunned by the powerful acid of [src]!</span>")
+		log_combat(user, target, "attacked", src)
 
-		M.adjust_blurriness(force/7)
+		target.adjust_blurriness(force/7)
 		if(prob(20))
-			M.Unconscious(force / 0.3)
-			M.Paralyze(force / 0.75)
-		M.drop_all_held_items()
+			target.Unconscious(force / 0.3)
+			target.Paralyze(force / 0.75)
+		target.drop_all_held_items()
```

Comment:
after doing a thing return true

Reply:
no see the problem with this is that if we return TRUE here when it succeeds, it won't reach afterattack, which means it won't lose petal when it DOES hit something.